pkg/storage/etcd/etcd_test_util.go

```diff
@@ -155,3 +160,32 @@ func NewEtcdTestClientServer(t *testing.T) *EtcdTestServer {
 	}
 	return server
 }
+
+// NewEtcdTestStorage creates a new storage.Interface and TestServer
+func NewEtcdTestStorage(t *testing.T, codec runtime.Codec, prefix string) (*EtcdTestServer, storage.Interface) {
+	server := NewEtcdTestClientServer(t)
+	storage := NewEtcdStorage(server.client, codec, prefix)
+	return server, storage
+}
+
+// CreateObj will create a single object using the storage interface
+func CreateObj(t *testing.T, helper storage.Interface, name string, obj, out runtime.Object, ttl uint64) error {
+	err := helper.Set(context.TODO(), name, obj, out, ttl)
+	if err != nil {
+		t.Errorf("Unexpected error %v", err)
+	}
+	return err
+}
+
+// CreatePodList will properly create a podlist using the storage interface
+func CreatePodList(t *testing.T, prefix string, helper storage.Interface, list *api.PodList) error {
```

Comment:
+1 for reusing generic infrastructure (I forgot about these functions)
Shouldn't something like that work:

func CreateList(t *testing.T, prefix string, helper storage.Interface, list runtime.Object) error {
  items, err := runtime.ExtractList(list)
  // handle error
  for i item := range items {
    returnedObj := reflect.New(reflect.TypeOf(item)).Interface().(runtime.Object)
    meta, err := meta.Accessory(item)
    // handle error
    err := CreateObj(t, helper, path.Join(prefix, meta.Name(), item, returnedObj, 0)
    // handle error
    items[i] = returnedObj
  }
  err := runtime.SetList(list, items)
  // handle error
  return nil
}